bundles/org.openhab.binding.tivo/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,153 @@
+				<default>31339</default>
+				<required>true</required>
+				<label>TCP Port</label>
+				<description>The
+					TCP port number used to connect to the TiVo.<![CDATA[ <b> ]]>Default:
+					31339<![CDATA[ </b> ]]></description>
+				<advanced>true</advanced>
+			</parameter>
+			<parameter name="numRetry" type="integer" max="20" min="0">
+				<default>5</default>
+				<required>true</required>
+				<label>Connection Retries</label>
+				<description>The
+					number of times to attempt reconnection to the TiVo box, if there is a connection failure.<![CDATA[ <b> ]]>Default:
+					5<![CDATA[ </b> ]]></description>
+				<advanced>true</advanced>
+			</parameter>
+			<parameter name="keepConActive" type="boolean">
+				<default>true</default>
+				<required>true</required>
+				<label>Keep Connection Open</label>
+				<description>Keep
+					connection to the TiVo open. Recommended for monitoring the TiVo for changes in TV channels. Disable if other
+					applications that use the Remote Control Protocol port will also be used e.g. mobile remote control applications.<![CDATA[ <b> ]]>Default:
+					True (Enabled)<![CDATA[ </b> ]]></description>
+			</parameter>
+			<parameter name="pollForChanges" type="boolean">
+				<default>true</default>
+				<required>true</required>
+				<label>Poll for Channel Changes</label>
+				<description>Check
+					TiVo for channel changes. Enable if openHAB and a physical remote control (or other services use the Remote Control
+					Protocol) will be used. <![CDATA[ <b> ]]>Default:
+					True (Enabled)<![CDATA[ </b> ]]></description>
+			</parameter>
+			<parameter name="pollInterval" type="integer" max="65535" min="3">
+				<default>10</default>
+				<required>true</required>
+				<label>Polling Interval (Seconds)</label>
+				<description>Number
+					of seconds between polling jobs to update status information from the TiVo. <![CDATA[ <b> ]]>Default:
+					10<![CDATA[ </b> ]]></description>
+			</parameter>
+			<parameter name="cmdWaitInterval" type="integer" max="500" min="0">
+				<default>200</default>
+				<required>true</required>
+				<label>Command Wait Interval (Milliseconds)</label>
+				<description>Period
+					to wait AFTER a command is sent to the TiVo in milliseconds, before checking that the command has completed. <![CDATA[ <b> ]]>Default:
+					200<![CDATA[ </b> ]]></description>
+				<advanced>true</advanced>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+	<channel-type id="channelForce">
+		<item-type>Number</item-type>
+		<label>Current Channel - Forced (FORCECH)</label>
+		<description>Displays the current channel number. When changed, tunes the DVR to the specified channel, cancelling any
+			recordings in progress if necessary i.e. all tuners are already in use / recording. The TiVo must be in Live TV mode
+			for this command to work. Type: Number (1-9999) [Decimals allowed for OTA sub-channels], DisplayFormat: %n</description>
+		<category>ChannelChange</category>
```

Comment:
This category doesn't exist. See https://www.openhab.org/docs/concepts/categories.html#channel-categories